client/src/components/visibility/Visibility.tsx

```diff
@@ -132,107 +173,152 @@ const VisibilityInput = ({
-              value={item.value}
-              disabled={isDisabled}
-              checked={visibility === item.value}
-              onChange={(e: ChangeEvent<HTMLInputElement>) =>
-                changeVisibility(e.target.value)
-              }
-              className={
-                markInvalid && !isDisabled
-                  ? "visibility-input--error"
-                  : "visibility-input"
-              }
-              data-cy={`visibility-${item.value}`}
-            />
-          </div>
-          <div
-            className={
-              isDisabled ? "cursor-not-allowed px-2" : "cursor-pointer px-2"
-            }
-            onClick={() => changeVisibility(item.value, isDisabled)}
-          >
-            <label
-              className={
-                isDisabled
-                  ? "cursor-not-allowed label-disabled"
-                  : "cursor-pointer"
-              }
-            >
-              {item.title}
-            </label>
-            <FontAwesomeIcon
-              icon={item.icon}
-              className={isDisabled ? "icon-disabled" : ""}
-            />
-          </div>
-        </div>
-        <div onClick={() => changeVisibility(item.value, isDisabled)}>
-          <InputHintLabel text={item.hint} />
-        </div>
-      </div>
+      <>
+        {disableByNamespaceOptions[namespaceVisibility]} If you need more
+        information about visibility, please check the{" "}
+        <ExternalLink
+          url={GitlabLinks.PROJECT_VISIBILITY}
+          role="text"
+          title="documentation"
+          className={cx(isTooltip && "link-rk-white")}
+        />
+        .
+      </>
+    );
+  };
+
+  const options = VISIBILITY_ITEMS.map((item) => {
+    const isDisabledByNamespace = visibilities.disabled.includes(item.value);
+    return (
+      <VisibilityInput
+        key={item.value}
+        item={item}
+        name={name}
+        isDisabled={disabled || isDisabledByNamespace}
+        changeVisibility={changeVisibility}
+        isChecked={visibility === item.value}
+        markInvalid={markInvalid}
+        tooltipContent={feedbackByNamespace(true)}
+      />
     );
   });
 
-  const disableByNamespaceOptions = {
-    public: "",
-    private:
-      "Public and Internal options are not available due to namespace restrictions",
-    internal: "Public is not available due to namespace restrictions",
-  };
-  const disabledByNamespace =
-    namespaceVisibility !== Visibilities.Public ? (
-      <InputHintLabel text={disableByNamespaceOptions[namespaceVisibility]} />
-    ) : null;
   const errorFeedback = markInvalid ? (
     <ErrorLabel text="Please select visibility" />
   ) : null;
 
   return (
     <>
-      <InputLabel text="Visibility" isRequired={isRequired} />
+      <legend>
+        {" "}
+        Visibility{" "}
+        {includeRequiredLabel ? (
+          <RequiredLabel isRequired={isRequired} />
+        ) : (
+          ""
+        )}{" "}
+      </legend>
```

Comment:
The legend renders with a large font size, you can style it with:

```suggestion
      <legend className="form-label fs-6">
        Visibility{" "}
        {includeRequiredLabel && <RequiredLabel isRequired={isRequired} />}
      </legend>
```